addons/README.md

```diff
@@ -2,8 +2,155 @@
 
 Pharos Cluster includes common functionality as addons. Addons can be enabled by introducing and enabling them in `cluster.yml`.
 
+- [Built-in Addons](#built-in-addons)
+- [Using External Addons](#using-external-addons)
+- [Creating an Addon](#creating-an-addon)
+
+## Built-in Addons
+
 * [Ingress NGINX](ingress-nginx.md)
 * [Cert Manager](cert-manager.md)
 * [Kured](kured.md)
 * [Kubernetes Dashboard](kube-dashboard.md)
 * [OpenEBS](openebs.md)
+
+## Using External Addons
+
+External addons are loaded by default from the `addons` folder which is located in the same path as `cluster.yml`. It's possible to load external addons from other paths by defining `addon_paths`. These paths are also relative to the `cluster.yml`.
+
+```yaml
+addon_paths:
+  - "./myaddons"
+```
+
+## Creating an Addon
+
+A Kontena Pharos cluster addon is a folder that contains `addon.rb` (Ruby DSL) and one or more Kubernetes YAML files in the `resources` folder. It is possible to use `erb` templating language if you use `.yml.erb` as the file extension.
+
+### Addon Template Variables
+
+Following variables are always available to resource templates:
+
+- `name` - addon name
+- `version` -  addon version
+- `license` - addon license
+- `arch` - cluster cpu architecture
+- `config` - user defined configuration variables
+
+### Validating Addon Configuration
+
+Validation rules can be defined in the `config_schema` block. Block uses [dry-validation](http://dry-rb.org/gems/dry-validation/) as the DSL.
+
+### Example
+
+`addons/hello-world/addon.rb`:
+```ruby
+Pharos.addon 'hello-world' do
+  version '1.0.0'
+  license 'Apache License 2.0'
+
+  # config schema is optional, it provides validation rules for user configurable options
+  config_schema {
+    required(:replicas).filled(:int?)
+    optional(:virtualhost).filled(:str?)
+    optional(:tls_secret).filled(:str?)
+  }
+
+  # install is optional, it can be used to customize the installation process if needed
+  #install {
+  #  logger.info "this happens before install"
+  #  install
```

Comment:
wouldn't this cause recursive loop? should the inner `install` be actually `apply_stack`?